### ISSUE TITLE
unauthorized does not return 403

### ISSUE THREAD

**Issue by boazin** [mattupstate/flask-security#403]
This is the code:

```
class MgmtList(Resource):
    @login_required
    @roles_required('admin')
    def get(self):
        #get and return mgmt list
```

Without `roles_required` I'm getting the list without authenticaltion
With `roles_required` I'm getting redirected to '/' instead of getting a 403.

Even tried `app.config['SECURITY_UNAUTHORIZED_VIEW'] = None`

I'm confused. What should I do to make the endpoint protected by authentication?

**Comment by jonafato** [comment]
This is a duplicated of #333, and the discussion there may be relevant to your use case. That said, the behavior here definitely does not match the documentation. The fix is fairly simple and would require checking for `None` in [this function](https://github.com/mattupstate/flask-security/blob/4049c0620383f42d37950c7a35af5ddd6df0540f/flask_security/decorators.py#L43-46).

**Comment by pradeeps** [comment]
I don't have this fix though I am using Flask-Security==1.7.5  ? Is this released ?

**Comment by jirikuncar** [comment]
closed via 78f63858
